### PR TITLE
V4.4.x

### PR DESCRIPTION
**Pull request by JamieVangeysel**

### CODE REVIEW

**Review by JamieVangeysel — COMMENTED** [review]
LGTM

File: src/controllers/upload.controller.ts [hunk 1]
```diff
@@ -98,28 +109,15 @@ export default async function (fastify: FastifyInstance) {
...
-
-            dt = new Date(Date.UTC(year, month - 1, day))
-          }
-        }
-
         let mimetype = data.mimetype
         let filename = data.filename
         let filesize = data.file.bytesRead
 
+        let dt = deleteOnDate(request.query.directory_id, path.parse(filename).name)
```
Comment (OhSoooLucky) [inline src/controllers/upload.controller.ts:116]: Should be a const

Comment (OhSoooLucky) [inline src/controllers/upload.controller.ts]: Should be a const

File: src/controllers/upload.controller.ts [hunk 2]
```diff
@@ -98,28 +109,15 @@ export default async function (fastify: FastifyInstance) {
...
-          if (/^[0-9]{8}$/.test(last8.toLocaleLowerCase())) {
-            const year = parseInt(last8.substring(0, 4))
-            const month = parseInt(last8.substring(4, 6))
-            const day = parseInt(last8.substring(6, 8))
-
-            dt = new Date(Date.UTC(year, month - 1, day))
-          }
-        }
-
         let mimetype = data.mimetype
```
Comment (OhSoooLucky) [inline src/controllers/upload.controller.ts:112]: Should be const variables

File: src/repositories/document.repository.ts [hunk 3]
```diff
@@ -38,7 +38,7 @@ export default class Document {
     r.input('user_id', sql.Int, user_id)
 
     let result
-    if (new RegExp('^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$').test(id.toString(10))) {
+    if (/^[{]?[0-9abcdefABCDEF]{8}-([0-9abcdefABCDEF]{4}-){3}[0-9abcdefABCDEF]{12}[}]?$/.test(id.toString(10))) {
```
Comment (OhSoooLucky) [inline src/repositories/document.repository.ts:41]: a-fA-F should work?

**Review by OhSoooLucky — CHANGES_REQUESTED** [review]

File: src/controllers/upload.controller.ts (hunk 2 above)
Comment (JamieVangeysel) [reply src/controllers/upload.controller.ts:112]: no

File: src/controllers/upload.controller.ts (hunk 1 above)
Comment (JamieVangeysel) [reply src/controllers/upload.controller.ts]: yes

**Review by OhSoooLucky — APPROVED** [review]

**Pull request merged**